src/main/java/com/contrastsecurity/maven/plugin/sdkx/DefaultContrastScanSDK.java

```diff
@@ -38,6 +40,46 @@ public DefaultContrastScanSDK(final ContrastSDK contrast, final String restURL)
     this.restURL = ContrastSDKUtils.ensureApi(Objects.requireNonNull(restURL));
   }
 
+  @Override
+  public Project createProject(final String organizationId, final CreateProjectRequest request)
+      throws IOException, UnauthorizedException {
+    // requests made with ContrastSDK.makeRequest must have their path prepended with "/"
+    final String path = String.join("/", "", "sast", "organizations", organizationId, "projects");
+    final String json = gson.toJson(request);
+    try (Reader reader =
+        new InputStreamReader(
+            contrast.makeRequestWithBody(HttpMethod.POST, path, json, MediaType.JSON))) {
+      return gson.fromJson(reader, Project.class);
+    }
+  }
+
+  @Override
+  public Project findProjectByName(final String organizationId, final String projectName)
+      throws IOException, UnauthorizedException {
+    // requests made with ContrastSDK.makeRequest must have their path prepended with "/"
+    final String uri =
+        String.join("/", "", "sast", "organizations", organizationId, "projects")
+            + "?unique=true&name="
+            + URLEncoder.encode(projectName, "US-ASCII");
```

Comment:
Can we use `StandardCharsets.US_ASCII` here? Also why not `UTF-8`?

Reply:
Actually, should we be using the Maven source encoding value: https://stackoverflow.com/questions/3017695/how-to-configure-encoding-in-maven?

I guess that’s the encoding for Java source though. Not sure it would apply to the project name in Maven XML.

Reply:
Right, the Maven source encoding is the encoding for the source files.

If I've read the HTTP specification correctly, then URIs in the request line use US-ASCII

We cannot use StandardCharsets.US_ASCII because that method overload was introduced in Java 10 https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/URLEncoder.html#encode(java.lang.String,java.nio.charset.Charset) 

I guess I can use

```java
URLEncoder.encode(projectName, StandardCharsets.US_ASCII.name());
```

Reply:
I've added a test for this

Reply:
👍 can you add a test case with a project name that includes unicode chars? I think that test case would fail since we’d translate all UTF-8 chars to `?` (or `%3F`). 

The JavaDoc for `URLEncoder.encode` recommends `UTF-8` for this reason I think: https://docs.oracle.com/javase/7/docs/api/java/net/URLEncoder.html#encode(java.lang.String,%20java.lang.String)

Reply:
OK we got to the bottom of this. It should definitely be UTF-8. Test added